pkg/sac/scope_checker_or.go

```diff
@@ -0,0 +1,212 @@
+	for _, checker := range s.scopeCheckers {
+		err := checker.PerformChecks(ctx)
+		// Short-circuit on the first non-error perform check.
+		if err != nil {
+			performChecksErrs = multierror.Append(performChecksErrs, err)
+		}
+	}
+	if performChecksErrs == nil {
+		return nil
+	}
+	return performChecksErrs
+}
+
+func (s orScopeChecker) TryAllowed(subScopeKeys ...ScopeKey) TryAllowedResult {
+	result := Deny
+	for _, checker := range s.scopeCheckers {
+		if res := checker.TryAllowed(subScopeKeys...); res == Allow {
+			return res
+		} else if res == Unknown {
+			result = Unknown
+		}
+	}
+	return result
+}
+
+func (s orScopeChecker) Allowed(ctx context.Context, subScopeKeys ...ScopeKey) (bool, error) {
+	var allowedErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		allowed, err := checker.Allowed(ctx, subScopeKeys...)
+		// Short-circuit on the first allowed check result.
+		if err != nil {
+			allowedErrs = multierror.Append(allowedErrs, err)
+		} else if allowed {
+			return allowed, nil
+		}
+	}
+	if allowedErrs != nil {
+		return false, allowedErrs
+	}
+	return false, nil
+}
+
+func (s orScopeChecker) TryAnyAllowed(subScopeKeyss [][]ScopeKey) TryAllowedResult {
+	result := Deny
+	for _, checker := range s.scopeCheckers {
+		if res := checker.TryAnyAllowed(subScopeKeyss); res == Allow {
+			return res
+		} else if res == Unknown {
+			result = Unknown
+		}
+	}
+	return result
+}
+
+func (s orScopeChecker) AnyAllowed(ctx context.Context, subScopeKeyss [][]ScopeKey) (bool, error) {
+	var anyAllowedErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		allowed, err := checker.AnyAllowed(ctx, subScopeKeyss)
+		// Short-circuit on the first allowed check result.
+		if err != nil {
+			anyAllowedErrs = multierror.Append(anyAllowedErrs, err)
+		} else if allowed {
+			return allowed, nil
+		}
+	}
+	if anyAllowedErrs != nil {
+		return false, anyAllowedErrs
+	}
+	return false, nil
+}
+
+func (s orScopeChecker) TryAllAllowed(subScopeKeyss [][]ScopeKey) TryAllowedResult {
+	result := Deny
+	for _, checker := range s.scopeCheckers {
+		if res := checker.TryAllAllowed(subScopeKeyss); res == Allow {
+			return res
+		} else if res == Unknown {
+			result = Unknown
+		}
+	}
+	return result
+}
+
+func (s orScopeChecker) AllAllowed(ctx context.Context, subScopeKeyss [][]ScopeKey) (bool, error) {
+	var allAllowedErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		allowed, err := checker.AllAllowed(ctx, subScopeKeyss)
+		// Short-circuit on the first allowed check result.
+		if err != nil {
+			allAllowedErrs = multierror.Append(allAllowedErrs, err)
+		} else if allowed {
+			return allowed, nil
+		}
+	}
+	if allAllowedErrs != nil {
+		return false, allAllowedErrs
+	}
+	return false, nil
+}
+
+func (s orScopeChecker) ForClusterScopedObject(obj ClusterScopedObject) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].ForClusterScopedObject(obj)
```

Comment:
Same as above (and below), this seems to change the current instance which is likely unintended.